.github/scripts/junit_to_summary.py

```
@@ -0,0 +1,106 @@
+        print("usage: junit_to_summary.py test-results.xml [max_lines]", file=sys.stderr)
+        return 0
+
+    path = Path(sys.argv[1])
+    max_lines = int(sys.argv[2]) if len(sys.argv) > 2 else MAX_LINES_DEFAULT
+
+    if not path.exists():
+        print("### Test results\n\n_No JUnit report produced._\n")
+        return 0
+
+    try:
+        root = ET.parse(path).getroot()
```

Ruff (0.15.12)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)